debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/snapshot/query/SelectAllSnapshotQuery.java

```diff
@@ -47,9 +52,33 @@ public Optional<String> snapshotQuery(String tableId, List<String> snapshotSelec
         final OracleOffsetContext offset = (OracleOffsetContext) snapshotContext.offset;
 
         final String snapshotOffset = offset.getScn().toString();
-        String columns = String.join(", ", snapshotSelectColumns);
+
+        TableId table = TableId.parse(tableId);
+        String columns = getColumns(snapshotContext.tables.forTable(new TableId(snapshotContext.catalogName, table.catalog(), table.table())));
         assert snapshotOffset != null;
         return Optional.of(String.format("SELECT %s FROM %s AS OF SCN %s", columns, tableId, snapshotOffset));
+    }
 
+    private String getColumns(Table table) {
+        StringBuilder columnsSb = new StringBuilder();
```

Comment:
```suggestion
        StringBuilder columnsList = new StringBuilder();
```
And update the usages elsewhere.